provider/datagen/src/transform/cldr/locale_canonicalizer/likely_subtags.rs

```diff
@@ -74,89 +99,156 @@ impl IterableDataProvider<LikelySubtagsForScriptRegionV1Marker> for crate::Datag
-            // Computes the delta of the entry and assigns to the pattern.
-            // Errors if the delta is not assignable to the pattern.
-            macro_rules! with_diff {
-                ($pat:pat => $stmt:expr ) => {
-                    if let $pat = (
-                        if entry.0.language != entry.1.language {
-                            entry.1.language
-                        } else {
-                            Language::UND
-                        },
-                        if entry.0.script != entry.1.script {
-                            entry.1.script
-                        } else {
-                            None
-                        },
-                        if entry.0.region != entry.1.region {
-                            entry.1.region
-                        } else {
-                            None
-                        },
-                    ) {
-                        $stmt;
-                    } else {
-                        panic!(
-                            "The expansion {:?} -> {:?} can not be stored in the pattern {}",
-                            entry.0,
-                            entry.1,
-                            stringify!($pat)
-                        );
-                    }
-                };
-            }
+struct LikelySubtagsResources<'a> {
+    likely_subtags: &'a cldr_serde::likely_subtags::Resource,
+    basic_plus_languages: HashSet<Language>,
+}
 
-            if !entry.0.language.is_empty() {
-                let lang = entry.0.language;
-                if let Some(script) = entry.0.script {
-                    with_diff!((Language::UND, None, Some(region)) => language_script.insert((lang.into(), script.into()), region));
-                } else if let Some(region) = entry.0.region {
-                    with_diff!((Language::UND, Some(script), None) => language_region.insert((lang.into(), region.into()), script));
-                } else {
-                    with_diff!((Language::UND, Some(script), Some(region)) => language.insert(lang.into(), (script, region)));
+impl<'a> LikelySubtagsResources<'a> {
+    fn try_from_source_data(
+        source_data: &'a SourceData,
+    ) -> Result<LikelySubtagsResources, DataError> {
+        let likely_subtags: &cldr_serde::likely_subtags::Resource = source_data
+            .cldr()?
+            .core()
+            .read_and_parse("supplemental/likelySubtags.json")?;
+        let coverage_levels: &cldr_serde::coverage_levels::Resource = source_data
+            .cldr()?
+            .core()
+            .read_and_parse("coverageLevels.json")?;
+        let basic_plus_languages = Self::get_basic_plus_languages(coverage_levels);
+        Ok(LikelySubtagsResources {
+            likely_subtags,
+            basic_plus_languages,
+        })
+    }
+
+    fn get_basic_plus_languages(
+        coverage_levels: &cldr_serde::coverage_levels::Resource,
```

Comment:
i assume this pulls the resolved coverage level? otherwise sublocales will be missed